pennylane/devices/execution_config.py

```diff
@@ -15,11 +15,28 @@
 Contains the :class:`ExecutionConfig` data class.
 """
 from dataclasses import dataclass
-from typing import Optional
+from typing import Optional, Union
 
 from pennylane.workflow import SUPPORTED_INTERFACES
 
 
+@dataclass
+class MCMConfig:
+    """A class to store mid-circuit measurement configurations."""
+
+    mcm_method: Optional[str] = None
+    """Which mid-circuit measurement strategy to use. Use ``deferred`` for the deferred
+    measurements principle and "one-shot" if using finite shots to execute the circuit
+    for each shot separately. If not specified, the device will decide which method to
+    use."""
+
+    postselect_mode: Optional[str] = None
+    """Configuration for handling shots with mid-circuit measurement postselection. If
+    ``"hw-like"``, invalid shots will be discarded and only results for valid shots will
+    be returned. If ``"fill-shots"``, results corresponding to the original number of
+    shots will be returned. If not specified, the device will decide which mode to use."""
+
```

Comment:
Might be worth adding a post init validation here too, just to provide better error messages if the string isn't correct.

Reply:
I wasn't familiar with this syntax, thanks for showing it here :)

I think I prefer to add validation with `__post_init__` for now as it is more consistent with `ExecutionConfig`. We can potentially come back to this in the future though.